packages/flutter/lib/src/material/tooltip.dart

```diff
@@ -0,0 +1,295 @@
+    super.initState();
+    _performance ??= new Performance(duration: config.fadeDuration)
+      ..addStatusListener((PerformanceStatus status) {
+        switch (status) {
+          case PerformanceStatus.completed:
+            assert(_entry != null);
+            assert(_timer == null);
+            resetShowTimer();
+            break;
+          case PerformanceStatus.dismissed:
+            assert(_entry != null);
+            assert(_timer == null);
+            _entry.remove();
+            _entry = null;
+            break;
+          default:
+            break;
+        }
+      });
+  }
+
+  void didUpdateConfig(Tooltip oldConfig) {
+    super.didUpdateConfig(oldConfig);
+    if (config.fadeDuration != oldConfig.fadeDuration)
+      _performance.duration = config.fadeDuration;
+    if (_entry != null &&
+        (config.message != oldConfig.message ||
+         config.backgroundColor != oldConfig.backgroundColor ||
+         config.style != oldConfig.style ||
+         config.textColor != oldConfig.textColor ||
+         config.borderRadius != oldConfig.borderRadius ||
+         config.height != oldConfig.height ||
+         config.padding != oldConfig.padding ||
+         config.opacity != oldConfig.opacity ||
+         config.verticalOffset != oldConfig.verticalOffset ||
+         config.screenEdgeMargin != oldConfig.screenEdgeMargin ||
+         config.preferBelow != oldConfig.preferBelow))
+      _entry.markNeedsBuild();
+  }
+
+  void resetShowTimer() {
+    assert(_performance.status == PerformanceStatus.completed);
+    assert(_entry != null);
+    _timer = new Timer(config.showDuration, hideTooltip);
+  }
+
+  void showTooltip() {
+    if (_entry == null) {
+      RenderBox box = context.findRenderObject();
+      Point target = box.localToGlobal(box.size.center(Point.origin));
+      _entry = new OverlayEntry(builder: (BuildContext context) {
+        TextStyle textStyle = (config.style ?? Theme.of(context).text.body1).copyWith(color: config.textColor ?? Colors.white);
+        return new _TooltipOverlay(
+          message: config.message,
+          backgroundColor: config.backgroundColor ?? Colors.grey[700],
+          style: textStyle,
+          borderRadius: config.borderRadius,
+          height: config.height,
+          padding: config.padding,
+          opacity: config.opacity,
+          performance: _performance,
+          target: target,
+          verticalOffset: config.verticalOffset,
+          screenEdgeMargin: config.screenEdgeMargin,
+          preferBelow: config.preferBelow
+        );
+      });
+      Overlay.of(context).insert(_entry);
+    }
+    _timer?.cancel();
+    if (_performance.status != PerformanceStatus.completed) {
+      _timer = null;
+      _performance.forward();
+    } else {
+      resetShowTimer();
+    }
+  }
+
+  void hideTooltip() {
+    assert(_entry != null);
+    _timer?.cancel();
+    _timer = null;
+    _performance.reverse();    
+  }
+
+  void deactivate() {
```

Comment:
Because if you move the Tooltip to another part of the tree while a tooltip is showing, you want the earlier tooltip to go away ASAP.